scala-thoughts.md

```diff
@@ -0,0 +1,51 @@
+#Scala Thoughts#
+
+###What's easy to do in Scala?  What's not?###
+
+Functional programming seems relatively easy in Scala.  In this respect, the
+language had many similarities to other functional languages like Racket and
+Haskell.  I didn't explicitly have to say "return", as is true of other functional
+languages.  The use of higher-order functions was similar to my experiences with
+Racket as well.  Scala also does support pattern matching like Haskell, but it
+seemed slightly more cumbersome.  This made it very easy to, for example, return
+different results based on different inputs, as we did in the `matchOnInputType`
+function.  It was also very easy to declare variables and use type inference, at
```

Comment:
I think it's interesting that you mentioned it's both easy to do functional programming and declare variables/type inference. I agree with the sentiment. However, most functional languages make it more cumbersome to declare variables through let statements. It's interesting to try to classify Scala in terms of imperative/functional because it can do both rather well.